api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -33,6 +33,16 @@
  */
 public interface Context
 {
+  /*
```

Comment:
All the point 3 was suggesting that the component to which the attribute applies does does not look for the attributes in different context. So there types can be as different as they can be.

It will not impact the behavior of the component a bit.